x-pack/platform/plugins/shared/task_manager/server/task_scheduling.ts

```diff
@@ -284,12 +284,23 @@ export class TaskScheduling {
    */
   public async ensureScheduled(
     taskInstance: TaskInstanceWithId,
-    options?: Record<string, unknown>
+    options?: ScheduleOptions
   ): Promise<TaskInstanceWithId> {
     try {
       return await this.schedule(taskInstance, options);
     } catch (err) {
       if (err.statusCode === VERSION_CONFLICT_STATUS) {
+        // check if task specifies a schedule interval
```

Comment:
I spent some time thinking about a way we could do this more generically (to allow for updates to more than just the schedule) but our task update functions are currently also very targeted (just enable/disable and update state and schedule) so to do a more would require a much larger change. I think we can wait to see if we get another request to update something during `ensureScheduled` to try to tackle a bigger refactor.